bcitflex/course.py

```diff
@@ -25,16 +25,32 @@ def filters_from_form(form: ImmutableMultiDict) -> ModelFilter:
 @bp.route("/")
 @bp.route("/courses", methods=["GET", "POST"])
 def index():
-    courses = DBSession.scalars(select(Course)).all()
+    page = request.args.get('page', 1, type=int)
+    per_page = 25
+
     if request.method == "POST":
         print(request.form)
-        courses = filters_from_form(request.form).filter(courses)
+        courses_query = filters_from_form(request.form).filter(DBSession.query(Course))
```

Comment:
Any particular reason to switch from `select` to `query`? As of SQLAlchemy 2.0 the [Query API has been marked as legacy](https://docs.sqlalchemy.org/en/20/orm/queryguide/query.html#legacy-query-api).

Reply:
@jonbiemond  `query` implements static method .paginate() from https://flask-sqlalchemy.palletsprojects.com/en/3.1.x/pagination/

there is no other implementation of SQL-Alchemy Pagination